libs/utils/utils.js

```diff
@@ -638,17 +638,32 @@ const decorateCopyLink = (a, evt) => {
   });
 };
 
+function convertStageLinks(config, anchors) {
+  if (config.env?.name === 'prod' || !config.stageDomainsMap) return;
+  const { hostname } = window.location;
+  const matchedRules = Object.entries(config.stageDomainsMap)
+    .find(([domain]) => hostname.includes(domain));
+  if (!matchedRules) return;
+  const [, domainsMap] = matchedRules;
+  [...anchors].forEach((a) => {
+    const matchedDomain = Object.keys(domainsMap)
+      .find((domain) => a.href.includes(domain));
+    if (!matchedDomain) return;
+    a.href = a.href.replace(a.hostname, domainsMap[matchedDomain] === 'origin'
+      ? hostname
+      : domainsMap[matchedDomain]);
+  });
+}
+
 export function decorateLinks(el) {
   const config = getConfig();
   decorateImageLinks(el);
   const anchors = el.getElementsByTagName('a');
+  convertStageLinks(config, anchors);
```

Comment:
At one point I remember we intentionally moved everything inside the existing `forEach` on L663 to avoid re-iterating over the same list of `a` elements. Is anything preventing you from sticking to that patterns?

I'd also be curious of how `convertStageLinks` interacts with the `localizeLink` and whether the link localization is maintained in all use-cases. If I'm on stage, on a Bacom page in German and I want to go to the Photoshop one, would I be redirected to the German version OOTB? It might be that we need to call `localizeLink` first, but you should check that hunch.

Reply:
> At one point I remember we intentionally moved everything inside the existing forEach on L663 to avoid re-iterating over the same list of a elements. Is anything preventing you from sticking to that pattern?

Reply:
> At one point I remember we intentionally moved everything inside the existing forEach on L663 to avoid re-iterating over the same list of a elements. Is anything preventing you from sticking to that pattern?

IMO the choice between these is a trade-off.
The current implementation:
Advantage: better for production by avoiding redundant checks on each `a` element.
Disadvantage: less efficient on stage due to iterating over `a` elements twice.

The alternative approach, moving the logic inside the current `a`'s `reduce()`:
Advantage: more efficient on stage by doing just 1 iteration over `a` elements.
Disadvantage: slight overhead on production by performing checks for each anchor.

Anyway, in both cases, the performance impact is minimal at most, but I chose the current approach because it is better for production and separates concerns.

> I'd also be curious of how convertStageLinks interacts with the localizeLink and whether the link localization is maintained in all use-cases

Did some testing around this and it works OOTB. The only problem is that the `.html` extension persists and that works only for production. Anyway, this `.html` problem is out of the scope of this ticket.